js/src/forum/components/DiscussionPage.js

```diff
@@ -88,6 +89,24 @@ export default class DiscussionPage extends Page {
   view() {
     const discussion = this.discussion;
 
+    let content;
+
+    if (discussion) {
+      const stream = new PostStream({ state: this.streamState });
```

Comment:
I don't understand the point of a state class if it's not accessible at all.

I still see stream.render()...  Isn't the point of the states to avoid stuff like this?

Reply:
The state class IS accessible, through app.current.streamState. It'll be moved out to a more appropriate location in the app.current refactor.

As per this part, if we can figure out how to get rid of the annoying event coming from PostStream, we can safely redo this with the utmost best practices.

Reply:
Oh, I see. This is the component. Yeah, the point of switching to states is also to avoid doing `component.render()`. Why can't we pass the event handler below through a prop instead of an event?

Also - the positionChanged method doesn't refer to DiscussionPage at all - it could perhaps be moved into the post stream state ? Not sure what the best approach is here in terms of extensibility.

Reply:
I passed the event handler for now. Moving around positionChanged isn't necessary right now, I'd rather do that later. It doesn't add THAT much clutter, and I'd prefer to keep stuff that changes url/title/etc in the page class for now.